src/libutil/logging.cc

```diff
@@ -68,6 +69,34 @@ class SimpleLogger : public Logger
 
 Verbosity verbosity = lvlInfo;
 
+typedef enum {
+    logFormatRaw,
+    logFormatJSON,
+    logFormatBar,
+    logFormatBarWithLogs,
+} LogFormat;
```

Comment:
Can you move most of this to `src/nix`? That way `src/libutil` only contains the abstract Logger class and the implementations are in `src/nix`.

Reply:
Or alternatively to `src/libmain`.

Reply:
I've moved most of it to `libmain`. The implementation of `SimpleLogger` is still in that file, and `JSONLogger` is also still in `libutil` (because it's strongly tied to `handleJSONLogMessage` and that one is used in libstore). Does that look good?